gems/smithy-client/lib/smithy-client/plugins/anonymous_auth.rb

```diff
@@ -14,9 +13,14 @@ class AnonymousAuth < Plugin
           AnonymousProvider.new
         end
 
-        option(:anonymous_auth_scheme) do |_config|
-          Smithy::Client::AuthSchemes::Anonymous.new
+        class Handler < Client::Handler
+          def call(context)
+            Smithy::Client::Signers::Anonymous.new.sign(context)
```

Comment:
You could go even further and delete signers too and just push the logic in here - but let's see if this prototype works first. Also this should also check for noAuth.